sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4456,6 +4456,20 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
         """.stripMargin),
       Seq(Row(2), Row(1)))
   }
+
+  test("SPARK-39448: Add ReplaceCTERefWithRepartition into nonExcludableRules list") {
+    withSQLConf(
+      SQLConf.OPTIMIZER_EXCLUDED_RULES.key -> ReplaceCTERefWithRepartition.ruleName) {
```

Comment:
I guess we can add `cte-xxx` like https://github.com/apache/spark/blob/master/sql/core/src/test/resources/sql-tests/inputs/cte.sql